app/Http/Controllers/TourController.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Tour;
+use Throwable;
+
+class TourController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $tours = Tour::with('images')->oldest()->paginate(config('app.default_paginate_tour'));
+        foreach ($tours as $tour) {
+            $images = $tour->images->first();
+            if (!$images) {
+                $arr[] = 'assets/images/destinations/NotFound.png';
+            } else $arr[] = $images->url;
+        }
+
+        return view('destinations', compact('tours', 'arr'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $tour = Tour::find($id);
+        if (!$tour) {
+            return redirect()->route('tours.index')->with('error', trans('messages.not_found_tour'));
+        }
+
+        $images = $tour->images->all();
+        if (!$images) {
```

Comment:
Cái $images dòng 63 trả về dữ liệu là mảng mà sao lại check trống bằng toán tử ! e? Nó có kiểm tra được mảng rỗng hay k?
Hơn nữa cái ảnh chỉ là 1 phần cua cái @tour, không có nó thì không hiện thôi chứ sao lại xử lý 
`return view('tour', compact('tour', 'img_fail'));` nếu không có $images?